src/main/groovy/org/transmartproject/batch/batchartifacts/MultipleItemsLineItemReader.groovy

```diff
@@ -0,0 +1,103 @@
+
+    {
+        setName(ClassUtils.getShortName(getClass()))
+    }
+
+    private static final String SAVED_CACHE = 'savedCache'
+    Resource resource
+    MultipleItemsFieldSetMapper<T> multipleItemsFieldSetMapper
+
+    private FlatFileItemReader<FieldSet> flatFileItemReader
+    private Collection<T> cache
+
+    @Override
+    T read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
+        if (!cache) {
+            FieldSet fieldSet = flatFileItemReader.read()
+            if (fieldSet != null) {
+                cache = multipleItemsFieldSetMapper.mapFieldSet(fieldSet)
+            }
+        }
+        if (cache) {
+            T item = cache[0]
+            cache.remove(item)
+            item
+        }
+    }
+
+    @Override
+    void afterPropertiesSet() throws Exception {
+        Assert.notNull(multipleItemsFieldSetMapper, 'mapper has to be specified')
+
+        DelimitedLineTokenizer lineTokenizer = new DelimitedLineTokenizer(
+                delimiter: DELIMITER_TAB
+        )
+        flatFileItemReader = new FlatFileItemReader(
+                lineMapper: new DefaultLineMapper(
+                        lineTokenizer: lineTokenizer,
+                        fieldSetMapper: new PassThroughFieldSetMapper(),
+                ),
+                recordSeparatorPolicy: new DefaultRecordSeparatorPolicy(),
+                resource: resource,
+                linesToSkip: 1,
+                skippedLinesCallback: new LineCallbackHandler() {
+                    @Override
+                    void handleLine(String line) {
+                        String[] header = new DelimitedLineTokenizer(DELIMITER_TAB)
+                                .tokenize(line).values
+                        Map uniqueHeaderNames = header.groupBy()
+                        if (header.length > uniqueHeaderNames.size()) {
+                            Set<String> violators = uniqueHeaderNames.findAll { it.value.size() > 1 }.keySet()
+                            throw new ValidationException("${violators} appear(s) more than once in the header.")
+                        }
+                        lineTokenizer.names = header
+                    }
+                },
+        )
+    }
+
+    @Override
+    void open(ExecutionContext executionContext) throws ItemStreamException {
+        cache = (Collection) executionContext.get(
+                getExecutionContextKey(SAVED_CACHE))
+        flatFileItemReader.open(executionContext)
+    }
+
+    @Override
+    void update(ExecutionContext executionContext) throws ItemStreamException {
+        executionContext.put(
+                getExecutionContextKey(SAVED_CACHE),
+                cache ? Lists.newLinkedList(cache) : null)
+        flatFileItemReader.update(executionContext)
+    }
+
+    @Override
+    void close() throws ItemStreamException {
+        flatFileItemReader.close()
+    }
+}
```

Comment:
I'd be more comfortable with a test testing restart in the middle of a step of this class, even though I can't see any obvious problem.